### PR TITLE
Get rid of RDD

### PR DESCRIPTION
**Pull request by hayssams**
## Summary
We get rid of the RDD everywhere except in the validators where we manipulate objects of type Any since we are having generic validators here.

**PR Type: Feature**

**Status: Ready to review**

**Breaking change? No**

### CODE REVIEW

File: src/main/scala/ai/starlake/job/ingest/IngestionJob.scala [hunk 1]
```diff
@@ -117,33 +91,34 @@ trait IngestionJob extends SparkJob {
...
+      errMessagesDS.take(100).foreach(rejected => logger.debug(rejected.replaceAll("\n", "|")))
     }
     val domainName = domain.name
     val schemaName = schema.name
 
     val start = Timestamp.from(Instant.now())
     val formattedDate = new java.text.SimpleDateFormat("yyyyMMddHHmmss").format(start)
 
-    if (settings.comet.sinkReplayToFile && !rejectedLinesRDD.isEmpty()) {
+    if (settings.comet.sinkReplayToFile && rejectedLinesDS.limit(1).count() > 0) {
```
Comment (cchepelov) [inline src/main/scala/ai/starlake/job/ingest/IngestionJob.scala:107]: a little sad to miss `isEmpty` on datasets. Is this due to back-compatiblity?

File: src/main/scala/ai/starlake/job/ingest/XmlIngestionJob.scala [hunk 2]
```diff
@@ -97,23 +94,23 @@ class XmlIngestionJob(
...
+        saveRejected(rejectedDS, session.emptyDataset[String])
         saveAccepted(
-          dataset,
           ValidationResult(
-            session.sparkContext.emptyRDD[String],
-            session.sparkContext.emptyRDD[String],
-            session.sparkContext.emptyRDD[Row]
+            session.emptyDataset[String],
+            session.emptyDataset[String],
+            dataset
```
Comment (cchepelov) [inline src/main/scala/ai/starlake/job/ingest/XmlIngestionJob.scala:110]: we had `emptyRDD[Row]` now we have `dataset`
Are we sure we're not unintentionally changing behaviour? 
If we're fixing some misbehaving thing, are we sure we now have a unit test to cover this?

Comment (cchepelov) [reply src/main/scala/ai/starlake/job/ingest/XmlIngestionJob.scala:110]: oh no, saw the context just before this hunk. Never mind.

File: src/main/scala/ai/starlake/job/ingest/IngestionJob.scala (hunk 1 above)
Comment (hayssams) [reply src/main/scala/ai/starlake/job/ingest/IngestionJob.scala:107]: Exactly.
For a customer you know very well. ```isEmpty``` appeared in Spark 2.4

**Pull request merged**